### PR TITLE
Support adding custom formats via server config. Add how-to guide with XDI as example.

### PR DESCRIPTION
**Pull request by danielballan**
The export function in the documentation is currently just a sketch. Once that is worked out more realistically, then I think this is ready to go.

### CODE REVIEW

**Comment by danielballan** [comment]
For convenience: https://github.com/XraySpectroscopy/XAS-Data-Interchange/blob/master/specification/spec.md

That's also linked in the docs already.

File: docs/source/how-to/custom-formats.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# Add Support for a Custom Format
+
+For a given "structure family" (e.g. `array`, `dataframe`) the Tiled server
+can provide the data---in whole or in part---in a variety of formats. 
+See {doc}`../tutorials/export` for a list of the formats supported out of the
+box.
+
+To teach Tiled to serve another format, first decide which structure family or
+families are appropriate for this format? For example, it make sense to export
```
Comment (dylanmcreynolds) [inline docs/source/how-to/custom-formats.md:9]: maybe remove the question mark?

Comment (danielballan) [reply docs/source/how-to/custom-formats.md:9]: Ah yes. Left over from a grammatical refactor.

File: docs/source/how-to/custom-formats.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    1. Use the standard [HTTP `Accept` Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept).
+
+    ```
+    $ curl -H 'Accept: text/csv' http://localhost:8000/dataframe/full/example
+    ```
+
+    2. Place the media type in a `format` query parameter.
+
+    ```
+    $ curl -H http://localhost:8000/dataframe/full/example?format=text/csv
```
Comment (kleinhenz) [inline docs/source/how-to/custom-formats.md:67]: Should be `curl -L ...`

Comment (danielballan) [reply docs/source/how-to/custom-formats.md]: Good catch. Was a copy/paste error. Fixed.

**Comment by danielballan** [comment]
@kleinhenz I reverted 918a768 and pushed an alternative implementation that achieves the same end in 75dbb99. See commit message in 75dbb99 for my reasoning.

I cobbled to together a very basic test in 613969a to verify it still works.

**Pull request merged**